src/utils/__tests__/log.test.ts

```diff
@@ -0,0 +1,29 @@
+import type { Logger } from "../log";
+import { logger } from "../log";
+
+describe("logger", () => {
+  const patternTest = [
+    ["INFO", "info"],
+    ["WARN", "warn"],
+    ["ERROR", "error"],
+    ["DEBUG", "debug"],
+    ["FATAL", "fatal"],
+  ];
+  it.each(patternTest)(
+    `should show the %s log when calling logger with %s level`,
+    (logDisplay, logLevel) => {
+      const consoleMock = jest
+        .spyOn(console, "error")
+        .mockImplementation(() => {
+          return true;
+        });
+
+      logger[logLevel as keyof Logger]("This is an example message.");
+
+      expect(consoleMock).toHaveBeenCalledTimes(1);
+      expect(consoleMock).toHaveBeenCalledWith(
+        expect.stringMatching(`${logDisplay}(.*) This is an example message.`),
```

Comment:
I don't know why the colon is not recognized correctly in Jest.
So I use `(.*)` instead.

Reply:
It's not because of the colon, but the color code like `[31m` or `[39m`.
It's tough to check the color code in this test, you can use `(.*)`.

Reply:
MUST:
Please make this match more strict.

Using spy, you can include datetime to our check.

```typescript
describe("logger", () => {
  const mockDate = new Date(0);
  const spy = jest.spyOn(global, "Date").mockImplementation(() => mockDate);

  afterAll(() => {
    spy.mockReset();
    spy.mockRestore();
  });

...
      expect(consoleMock).toHaveBeenCalledWith(
        expect.stringMatching(
          new RegExp(
            `\\[1970-01-01T00:00:00.000Z] (.*)${logDisplay}(.*): This is an example message.`,
          ),
        ),
```